src/services/db/GitFileSystemService.ts

```diff
@@ -433,9 +427,104 @@ export default class GitFileSystemService {
     })
   }
 
-  // TODO: Creates either directory or file
-  // ResourceDirectoryService used this to create a directory + file at the same time
-  create() {}
+  // Creates a file and the associated directory if it doesn't exist
+  create({
+    repoName,
+    userId,
+    content,
+    directoryName,
+    fileName,
+  }: {
+    repoName: string
+    userId: string
+    content: string
+    directoryName: string
+    fileName: string
+  }): ResultAsync<
+    { sha: string },
```

Comment:
This should be `GitCommitResult`.

Reply:
Changed in https://github.com/isomerpages/isomercms-backend/pull/892/commits/b0b811aec0ce097f1595fbfd772bb64d271c09da